src/Microsoft.AspNetCore.Mvc.Core/Internal/FileContentResultExecutor.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class FileContentResultExecutor : FileResultExecutorBase
+    {
+        public FileContentResultExecutor(ILoggerFactory loggerFactory)
+            : base(CreateLogger<FileContentResultExecutor>(loggerFactory))
+        {
+        }
+
+        public Task ExecuteAsync(ActionContext context, FileContentResult result)
+        {
+            SetHeaders(context, result);
+            _logger.FileResultExecuting(result.FileDownloadName);
```

Comment:
This is always done immediately after `SetHeaders()`. Suggest doing it there. Might be worth renaming that method to `SetHeadersAndLog()` but up to you.